src/libpmem2/mover.c

```diff
@@ -200,25 +211,51 @@ mover_delete(struct vdm *dms)
 	free((struct data_mover *)dms);
 }
 
+/*
+ * pmem2_future_detect_properties -- identifies how the future should behave
+ * depending on the properties of the underlying memory map and supported vdm
+ * features.
+ */
+static void
+pmem2_future_detect_properties(struct pmem2_map *map,
+	uint64_t *vdm_flags, bool *needs_flushing)
+{
+	enum pmem2_granularity gran = pmem2_map_get_store_granularity(map);
+	bool durable = vdm_is_supported(map->vdm, VDM_F_MEM_DURABLE);
+
+	switch (gran) {
+		case PMEM2_GRANULARITY_BYTE:
+			*needs_flushing = 0;
+			*vdm_flags = 0;
+			break;
+		case PMEM2_GRANULARITY_PAGE:
```

Comment:
What DSA do in case of non-pmem? I trought that it will write to the page buffer - and in this case, we don't need the durable flag, and we need a msync(persist)

Reply:
I was testing with sync pmem2 mover, and that is broken for nonpmem right now... I'll create a separate patch and then fix it here...

Reply:
Done.